xarray/backends/pydap_.py

```diff
@@ -209,7 +210,14 @@ class PydapBackendEntrypoint(BackendEntrypoint):
     url = "https://docs.xarray.dev/en/stable/generated/xarray.backends.PydapBackendEntrypoint.html"
 
     def guess_can_open(self, filename_or_obj: T_PathFileOrDataStore) -> bool:
-        return isinstance(filename_or_obj, str) and is_remote_uri(filename_or_obj)
+        if not (isinstance(filename_or_obj, str) and is_remote_uri(filename_or_obj)):
+            return False
+
+        # Check file extension to avoid claiming non-OPeNDAP URLs (e.g., remote Zarr stores)
+        _, ext = os.path.splitext(filename_or_obj.rstrip("/"))
+        # Pydap handles OPeNDAP endpoints, which typically have no extension or .nc/.nc4
+        # Reject URLs with non-OPeNDAP extensions like .zarr
+        return ext not in {".zarr", ".zip", ".tar", ".gz"}
```

Comment:
Not 100% sure on this. We could go further and require "dap" to be in the URL

Reply:
I don't think there's a standard extension for OpenDAP URLs. @Mikejmnez do you know?

Reply:
Well, actually, Thredds (TDS) does have this "standard" way to specify the protocol that may help to discern between opendap url vs non-opendap url: a TDS dap2 url will have a `dodsC` in its urls. A TDS dap4 url will have a `dap4` in its url. (see [here](https://pydap.github.io/pydap/en/faqs/dap2_or_dap4_url.html)). However, an  organization running an opendap server may decide how their own urls are exposed.